sdk/storage/Azure.Storage.DataMovement/MigrationGuide.md

```diff
@@ -154,6 +301,162 @@ TODO
 - Error reporting
 - Pause/Resume
 
+### Copy
+
+#### Copy blob to blob
+
+Note: The modern data movement library only supports service side sync copy.
+
+**Legacy:**
+```csharp
+// these values provided by your code
+string srcContainerName, srcBlobName, dstContainerName, dstBlobName;
+CloudBlobClient client;
+```
+```csharp
+// copy blob
+await TransferManager.DownloadAsync(
+    client.GetContainerReference(srcContainerName).GetBlockBlobReference(srcBlobName),
+    client.GetContainerReference(dstContainerName).GetBlockBlobReference(dstBlobName),
+    CopyMethod.ServiceSideSyncCopy);
+```
+**Modern:**
+```csharp
+// these values provided by your code
+string srcBlobUri, dstBlobUri;
+BlobsStorageResourceProvider blobs;
+TransferManager transferManager;
+```
+```csharp
+// copy blob
+TranferOperation operation = await transferManager.StartTransferAsync(
+    blobs.FromBlob(srcBlobUri),
+    blobs.FromBlob(dstBlobUri));
+await operation.WaitForCompletionAsync();
+```
+
+#### Copy share file to blob
+
+Note: File shares requires the Azure.Storage.DataMovement.Files.Shares package.
+
+**Legacy:**
+```csharp
+// these values provided by your code
+string containerName, blobName, shareName, filePath;
+CloudBlobClient blobClient;
+CloudFileClient fileClient;
+```
+```csharp
+// copy file
+await TransferManager.DownloadAsync(
+    blobClient.GetContainerReference(srcContainerName).GetBlockBlobReference(srcBlobName),
+    fileClient.GetShareReference(dstContainerName).GetRootDirectoryReference().GetBlockBlobReference(dstBlobName),
+    CopyMethod.ServiceSideSyncCopy);
+```
+**Modern:**
+```csharp
+// these values provided by your code
+string blobUri, fileUri;
+BlobsStorageResourceProvider blobs;
+ShareFilesStorageResourceProvider files;
+TransferManager transferManager;
+```
+```csharp
+// copy file
+TranferOperation operation = await transferManager.StartTransferAsync(
+    blobs.FromBlob(blobUri),
+    files.FromFile(fileUri));
+await operation.WaitForCompletionAsync();
+```
+
+### Progress Reporting
+
+#### Post-transfer report
+
+In the legacy data movement library, it was possible to check a report of how many files
+were or were not transferred in a directory transfer, as shown below. This is no
+longer supported in the modern library. Applications will need to enable progress reporting
```

Comment:
Maybe instead of saying it's no longer supported, maybe something along the lines of in the modern library you have to opt-in and enable progress reporting through `ProgressHandlerOptions` to make it sound more like, we do support this scenario still, it just shows up very differently.